File: src/_02_StructuralPattern/_02_01_Adapter/java/after/AccountUserDetail.java
package _02_StructuralPattern._02_01_Adapter.java.after;

import _02_StructuralPattern._02_01_Adapter.java.after.security.UserDetail;

public class AccountUserDetail implements UserDetail {

  private Account account;

  public AccountUserDetail(Account account) {
    this.account = account;
  }
  
  @Override
  public String getUsername() {
    return account.getName();
  }

  @Override
  public String getPassword() {
    return account.getPassword();
  }
  
}

File: src/_02_StructuralPattern/_02_01_Adapter/AccountUserDetail.js
import UserDetail from "./UserDetail.js";

/**
 * @typedef { import("./Account").default } Account
 */

export default class AccountUserDetail extends UserDetail {
  /** @type { Account } */
  #account;

  /** @param { Account } account */
  constructor(account) {
    super();
    this.#account =account;
  }

  /**
   * @override
   * @returns { string }
   */
  getUsername() {
    return this.#account.getName();
  }

  /**
   * @override
   * @returns { string }
   */
  getPassword() {
    return this.#account.getPw();
  }
}